pkg/push/api/api.go

```diff
@@ -561,7 +569,136 @@ func (s *PushService) UpdatePush(
 		}
 		return nil, dt.Err()
 	}
-	return &pushproto.UpdatePushResponse{}, nil
+
+	if updatedPushPb != nil {
+		// For security reasons we remove the service account from the API response
+		updatedPushPb.FcmServiceAccount = ""
+	}
+	return &pushproto.UpdatePushResponse{
+		Push: updatedPushPb,
+	}, nil
+}
+
+func (s *PushService) updatePushNoCommand(
+	ctx context.Context,
+	req *pushproto.UpdatePushRequest,
+	localizer locale.Localizer,
+	editor *eventproto.Editor,
+) (*pushproto.UpdatePushResponse, error) {
+	if err := s.validateUpdatePushRequestNoCommand(ctx, req, localizer); err != nil {
+		return nil, err
+	}
+	var updatedPushPb *pushproto.Push
+	var updatePushTagsEvent *eventproto.Event
+	var updatePushNameEvent *eventproto.Event
+	tx, err := s.mysqlClient.BeginTx(ctx)
+	if err != nil {
+		s.logger.Error(
+			"Failed to begin transaction",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+			)...,
+		)
+		dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalize(locale.InternalServerError),
+		})
+		if err != nil {
+			return nil, statusInternal.Err()
+		}
+		return nil, dt.Err()
+	}
+	err = s.mysqlClient.RunInTransaction(ctx, tx, func() error {
+		pushStorage := v2ps.NewPushStorage(tx)
+		push, err := pushStorage.GetPush(ctx, req.Id, req.EnvironmentNamespace)
+		if err != nil {
+			return err
+		}
+		prev := &domain.Push{}
+		if err = copier.Copy(prev, push); err != nil {
+			return err
+		}
+		push.Name = req.Name
+		push.Tags = req.Tags
```

Comment:
We can't update it like this because the console will only send the changed values. This way, when the user updates the Push name, it will delete the tags and vice-versa.

You must create a new function in the domain layer to update only the changed fields as follows.
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/feature/domain/feature.go#L957-L1043

Please take a look at this API for reference.
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/feature/api/feature.go#L666-L840

Reply:
Nice catch!
Yes, it should be fixed.
I found another thing that got my attention why it is listing all flags instead of getting the feature by ID directly [here](https://github.com/bucketeer-io/bucketeer/blob/main/pkg/feature/api/feature.go#L718-L724)

Reply:
We could just call this interface [GetFeature](https://github.com/bucketeer-io/bucketeer/blob/main/pkg/feature/storage/v2/feature.go#L202).

Reply:
I see, oke I will update this update tags logic and resolve problem with feature in another pr